src/AccessibilityInsights.Extensions.TelemetryTests/OSHelpersUnitTests.cs

```diff
@@ -1,57 +1,82 @@
+            Assert.AreEqual(1, registry.QueriesMade);
         }
 
         [TestMethod]
-        public void GetCurrentWindowsVersionForTelemetry_CurrentBuildDoesNotExist_ReturnsCurrentVersion()
+        public void GetVersion_CurrentBuildDoesNotExist_ReturnsCurrentVersion()
         {
-            using (ShimsContext.Create())
+            Tuple<string, string, string>[] expectedCalls =
             {
-                List<string> returnValues = new List<string> { "99.88", "" };
-                int index = 0;
-                ShimRegistry.GetValueStringStringObject = (_, __, def) => (++index > returnValues.Count ? def : returnValues[index - 1]);
+                MakeVersionTuple(CurrentVersion, "99.88"),
+                MakeVersionTuple(CurrentBuild, ""),
+            };
+            TestRegistry registry = new TestRegistry(expectedCalls);
 
-                Assert.AreEqual("99.88", OSHelpers.GetVersion());
-                Assert.AreEqual(2, index);
-            }
+            Assert.AreEqual("99.88", OSHelpers.GetVersion(registry.GetStringValue));
+            Assert.AreEqual(2, registry.QueriesMade);
         }
 
         [TestMethod]
-        public void GetCurrentWindowsVersionForTelemetry_AllDataExists_ReturnsCurrentVersionDotCurrentBuild()
+        public void GetVersion_AllDataExists_ReturnsCurrentVersionDotCurrentBuild()
         {
-            using (ShimsContext.Create())
+            Tuple<string, string, string>[] expectedCalls =
             {
-                List<string> returnValues = new List<string> { "99.88", "77" };
-                int index = 0;
-                ShimRegistry.GetValueStringStringObject = (_, __, def) => (++index > returnValues.Count ? def : returnValues[index - 1]);
+                MakeVersionTuple(CurrentVersion, "99.88"),
+                MakeVersionTuple(CurrentBuild, "77"),
+            };
+            TestRegistry registry = new TestRegistry(expectedCalls);
+
+            Assert.AreEqual("99.88.77", OSHelpers.GetVersion(registry.GetStringValue));
+            Assert.AreEqual(2, registry.QueriesMade);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void GetVersion_LiveData_ReturnsDataInCorrectFormat()
+        {
+            string version = OSHelpers.GetVersion();
+            string[] pieces = version.Split('.');
+            Assert.AreEqual(3, pieces.Length);
+            int majorVersion = int.Parse(pieces[0]);
+            int minorVersion = int.Parse(pieces[1]);
+            int build = int.Parse(pieces[2]);
 
-                Assert.AreEqual("99.88.77", OSHelpers.GetVersion());
-                Assert.AreEqual(2, index);
-            }
+            // Validation here assumes a range of Windows OS versions.
+            // Our lower bound is 6.1.7601  (Win 7 SP1)
+            // Our upper bound is 6.3.99999
+            // At current Win10 cadence, that will last until about 2040
```

Comment:
I wanted to strike a balance between unexpected changes in how Windows writes the version and creating a maintenance headache. I'll be astonished if this code is still running in 2020.